src/Tools/AnalyzerRunner/Options.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AnalyzerRunner
+{
+    internal class Options
+    {
+        public bool ConcurrentAnalysis;
+        public bool ReportSuppressedDiagnostics;
+        public HashSet<string> AnalyzerIds;
+        public bool Stats;
+        public string AnalyzerPath;
+        public string SolutionPath;
+        public bool UseAll;
+        public bool ApplyChanges;
+        public bool CodeFixes;
+        public bool FixAll;
+        public string LogFileName;
+
+        private Options() { }
+
+        internal static Options Create(string[] args)
+        {
+            // A valid call must have at least one parameter (a solution file). Optionally it can include /all or /id:SAXXXX.
+            if (args.Length < 1)
+            {
+                Utilities.PrintHelp();
+                return null;
+            }
+
+            var requiredArguments = args.Where(i => !i.StartsWith("/", StringComparison.Ordinal)).ToArray();
+            if (requiredArguments.Length < 2)
+            {
+                Utilities.WriteLine("The application requires the following arguments: analyzer assembly path and solution path.", ConsoleColor.Red);
+                return null;
+            }
+
+            var options = new Options()
```

Comment:
Roslyn's pattern seems ctor with readonly rather than object creation with object initializer.

Reply:
Also this is not the best way to process command line arguments -- it doesn't report errors for incorrect arguments. It's easy to make a typo and wonder what's going on. See https://github.com/dotnet/symreader-converter/blob/master/src/Pdb2Pdb/Pdb2Pdb.cs#L63 for example.